src/GitHub.Exports/Models/RepositoryModelBase.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Diagnostics;
+using GitHub.Primitives;
+using GitHub.UI;
+
+namespace GitHub.Models
+{
+    /// <summary>
+    /// The base class for local and remote repository models.
+    /// </summary>
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class RepositoryModelBase : NotificationAwareObject, IRepositoryModel
+    {
+        UriString cloneUrl;
+        Octicon icon;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RepositoryModelBase"/> class.
+        /// </summary>
+        /// <param name="name">The repository name.</param>
+        /// <param name="cloneUrl">The repository's clone URL.</param>
+        protected RepositoryModelBase(
+            string name,
+            UriString cloneUrl)
+        {
+            Name = name;
+            this.cloneUrl = cloneUrl;
+        }
+
+        /// <summary>
+        /// Gets the name of the repository.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// Gets the repository clone URL.
+        /// </summary>
+        public UriString CloneUrl
+        {
+            get { return cloneUrl; }
+            set
+            {
+                if (cloneUrl != value)
+                {
+                    cloneUrl = value;
+                    RaisePropertyChanged();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the name of the owner of the repository, taken from the clone URL.
+        /// </summary>
+        public string Owner => CloneUrl.Owner;
```

Comment:
CloneUrl can actually be null because we aren't doing null checks anywhere. Technically it should never be null right now because we earlier filter out repos that don't have remotes set, but in the future this assumption may change.
Might be a good idea to add nullchecks to the constructor for now, to document assumptions... or make sure this doesn't blow up if we don't have null checks there.